test/test_dh.c

```diff
@@ -340,7 +340,65 @@ int test_dh_pgen_pkey(ENGINE *e, void *data)
     return err;
 }
 
-int test_dh_pkey(ENGINE *e, void *data)
+#if !defined(WE_SINGLE_THREADED) && defined(_WIN32)
+
+typedef struct {
+    ENGINE* e;
+    EVP_PKEY* params;
+} DH_KEYGEN_THREAD_VARS;
+
+static DWORD WINAPI DhKeyGenThreadFunc(LPVOID arg)
+{
+    DH_KEYGEN_THREAD_VARS* vars = (DH_KEYGEN_THREAD_VARS*)arg;
+
+    return test_dh_pgen_pkey(vars->e, vars->params);
+}
+
+int test_dh_key_gen_multithreaded(ENGINE* e, EVP_PKEY* params)
+{
+    DH_KEYGEN_THREAD_VARS vars;
+    DH_KEYGEN_THREAD_VARS* pDataArray[1];
```

Comment:
I know this is cruft from my initial pass on this test, but I think since these are just arrays of size 1, they need not be arrays at all.

Reply:
I agree.  Changes have been made and committed.